webapp/packages/plugin-connections/src/ConnectionForm/DriverProperties/ConnectionDriverPropertiesTabService.ts

```diff
@@ -80,6 +80,26 @@ export class ConnectionDriverPropertiesTabService extends Bootstrap {
     const config = contexts.getContext(connectionConfigContext);
 
     config.properties = { ...state.config.properties };
+
+    if (config.driverId) {
+      const driver = this.dbDriverResource.get(config.driverId);
+
+      const driverMap = driver?.driverProperties?.reduce((acc, item) => {
+        acc.set(item.id, item.defaultValue);
+        return acc;
+      }, new Map());
+
+      for (const key in config.properties) {
+        // if it is default driver property (not custom)
+        if (driverMap?.has(key)) {
+          continue;
+        }
+
+        const value = config.properties[key];
+        delete config.properties[key];
+        config.properties[key.trim()] = typeof value === 'string' ? value.trim() : value;
+      }
```

Comment:
it's not recommended to modify current object in `for in` operator. You can create new object and add all properties and trim key & value if needed:
```js
      const trimmedProperties = { };
      for (const key in config.properties) {
        // if it is default driver property (not custom)
        if (driverMap?.has(key)) {
          trimmedProperties[key] = config.properties[key];
        }

        const value = config.properties[key];
        trimmedProperties[key.trim()] = typeof value === 'string' ? value.trim() : value;
      }
      config.properties = trimmedProperties;
```